### PR TITLE
feat: Add client interface

### PR DESCRIPTION
**Pull request by dojusa**
This will provide more flexibility to users to swap their impl for testing purposes, for example.

- feat: add client interface
- feat: make client impl private

### CODE REVIEW

**Review by antonmoller — APPROVED** [review]
Great change! 🚢 🇮🇹

File: client_charges.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ type StartChargeResponse struct {
 }
 
 // ListCharges to retrieve your charge points.
```
Comment (CodingBoulderingRepeat) [inline client_charges.go:64]:
```suggestion
// ListCharges to retrieve your charges.
```

File: client_charges.go [hunk 2]
```diff
@@ -107,13 +107,13 @@ func (c *Client) StartCharge(ctx context.Context, request *StartChargeRequest) (
 }
 
 // StopCharge stop a charge.
```
Comment (CodingBoulderingRepeat) [inline client_charges.go:109]: super nit
```suggestion
// StopCharge stops a charge.
```

File: client_charges.go [hunk 3]
```diff
@@ -107,13 +107,13 @@ func (c *Client) StartCharge(ctx context.Context, request *StartChargeRequest) (
...
-func (c *Client) StopCharge(ctx context.Context, chargeID int64) (*Charge, error) {
+func (c *clientImpl) StopCharge(ctx context.Context, chargeID int64) (*Charge, error) {
 	path := fmt.Sprintf("/v1/charges/%d/stop", chargeID)
 	return doPost[Charge](ctx, c, path, nil)
 }
 
 // RestartCharge restart or start a reserved charge.
```
Comment (CodingBoulderingRepeat) [inline client_charges.go:115]: super nit
```suggestion
// RestartCharge restarts or starts a reserved charge.
```

**Review by CodingBoulderingRepeat — APPROVED** [review]
LGTM

File: client_charges.go (hunk 1 above)
Comment (dojusa) [reply client_charges.go]: good catch!

**Pull request merged**